### ISSUE TITLE
Snippet does not have a typescript definition for publishTime.

### ISSUE THREAD

**Issue by Renari** [googleapis/google-api-nodejs-client#2366]
#### Environment details

  - OS: W10
  - Node.js version: v12.18.3
  - npm version: 6.14.6
  - `googleapis` version: 59.0.0

#### Steps to reproduce

  1. Do a search request like so:
```
youtube.search
      .list({
        part: ['snippet,id'],
        channelId: "<SOME CHANNEL ID>",
        maxResults: 25,
        order: 'date',
        type: ['video'],
        publishedAfter: date,
        key: "<YOUR API KEY>",
      })
      .then(response => {
        if (videos.items)
          console.log(videos.items[0].snippet?.publishTime)
      })
```
  2. Property 'publishTime' does not exist on type 'Schema$SearchResultSnippet', however it exists in the JSON data.

**Comment by sofisl** [comment]
Hi there! Thanks so much for the issue report! I've been able to reproduce the bug. 

This library is actually just auto-generated from a set of definitions, and this is an issue with the API itself. Could I trouble you to create this issue in this [public tracker](https://issuetracker.google.com/issues/new?component=186600&template=874803&pli=1)? Normally, I would post this issue on your behalf, but this method allows you to track the issue progress yourself. If you'd prefer not to post it, feel free to let me know and I will post it internally. Once this bug has been resolved, this library will be updated automatically :)
